### DISCUSSION TITLE
Material property 'A', requested by 'mobility' is not defined on block 0

### DISCUSSION THREAD

**Discussion by Johnney12138** [idaholab/moose#23400] [Q&A General] [answered]
Hello,
I was trying to solve a phase field problem. After running, it gave me an error message:
` *** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".
Material property 'A', requested by 'mobility' is not defined on block 0
Material property 'A', requested by 'mobility_face' is not defined on block 0
Material property 'A', requested by 'mobility_neighbor' is not defined on block 0`

The complete code is attached as follows:

`[Mesh]
    type = GeneratedMesh
    dim = 2               
    nx = 10               # Number of elements in the X direction.
    ny = 10               # Number of elements in the Y direction.
    xmin = 2
    xmax = 4              # Upper X Coordinate of the generated mesh.
    ymin = 2
    ymax = 4
    uniform_refine = 2
[]

[GlobalParams]
    derivative_order = 4
[]

[Variables]
    [./w]       # concentration
    [../]
    [./phi]   # phase field
    [../]
    [./T]   # temperature
    [../]
[]

[ICs]
    [./w]
      type = ConstantIC
      variable = w
      value = 0.3
    [../]
    [./T]
      type = ConstantIC
      variable = T
      value = 2.1
    [../]
    [./phi]
      type = SmoothCircleIC
      variable = phi
      x1 = 3
      y1 = 3
      invalue = 1
      outvalue = 0
      radius = 0.1
      int_width = 0.01
    [../]
[]

[Kernels]
    [./phi_dot]
      type = SusceptibilityTimeDerivative
      variable = phi
      f_name = tau
    [../]
    [./free_energy]
      type = AllenCahn
      variable = phi
      f_name = Fr
    [../]  
    [./anisoACinterface1]
      type = ACInterfaceKobayashi1
      variable = phi
    [../]
    [./anisoACinterface2]
      type = ACInterfaceKobayashi2
      variable = phi
    [../]
    [./w_dot]
      type = SusceptibilityTimeDerivative   
      variable = w
      f_name = chi                          
      coupled_variables = 'phi'
    [../]
    [./Diffusion]
      type = MatDiffusion                   
      variable = w
      diffusivity = Dchi                    
      coupled_variables = 'phi'
    [../]
    [./coupled_etaa0dot]
      type = CoupledSusceptibilityTimeDerivative
      variable = w
      v = phi
      f_name = ft
      coupled_variables = 'phi'
    [../]
    [./coupled_antitrapping]
      type = AntitrappingCurrent
      variable = w
      v = phi
      f_name = rhodiff
    [../]
    [./coupled_etaa0dot_time]
      type = TimeDerivative
      variable = T
    [../]
    [./coupled_etab0dot_diff]
      type = CoefDiffusion
      variable = T
      coef = 1e8
    [../]
    [./coupled_etab0dot_ctim]
      type = CoefCoupledTimeDerivative
      variable = T
      v = phi
      coef = -0.5
    [../]    
[] 

[Materials]
    [./A]
      type = InterfaceOrientationMaterial
      op = phi
      anisotropy_strength = 0.04
      mode_number = 4
      outputs = exodus
    [../]
    [./speexpress]
      type = ParsedMaterial           
      coupled_variables = 'phi w T'
      f_name = Fr
      material_property_names = 'lambda MCi'
      function = 'phi^3-phi+lambda*(1-phi^2)^2*(MCi*w+T)'
      output=exodus 
    [../]
    [./mobility]
      type = ParsedMaterial
      coupled_variables = 'w'  
      f_name = tau
      material_property_names = 'tau0 Le MCi k A'
      expression = 'tau0*A^2*(1/Le+MCi*(1+(1-k)*w))' 
      output=exodus
    [../]
    [./chemp]
      type = ParsedMaterial
      coupled_variables = 'phi'
      f_name = chi
      material_property_names = 'k'
      function = '1+k-(1-k)*phi' 
      output=exodus
    [../]
    [./matdiff]
      type = ParsedMaterial
      coupled_variables = 'phi'
      f_name = Dchi
      material_property_names = 'D'
      function = 'D*(1-phi)' 
      output=exodus
    [../]
    [./antitrap]
      type = ParsedMaterial
      coupled_variables = 'w'
      f_name = rhodiff
      material_property_names = 'k wid'
      function = 'wid/sqrt(2)*(1+(1-k)*w)' 
      output=exodus
    [../]
    [./int]
      type = ParsedMaterial
      coupled_variables = 'w'
      f_name = ft
      material_property_names = 'k'
      function = '1+(1-k)*w' 
      output=exodus
    [../]
    [./const]
      type = GenericConstantMaterial
      prop_names =  'tau0   Le  MCi   k  lambda   D  wid L erp'
      prop_values = '1e-5  1.0  0.15  0.15  3.1913  2.0  0.01 1 0.01'
    [../]
[] 

[Preconditioning]
    [./SMP]
      type = SMP        
      full = true      
    [../]
[]
 
[Executioner]
    type = Transient
    scheme = bdf2           
    solve_type = PJFNK     
    petsc_options_iname = '-pc_type -pc_hypre_type -ksp_gmres_restart'
    petsc_options_value = 'hypre    boomeramg      31'
    l_tol = 1.0e-3          
    l_max_its = 100          
    nl_max_its = 100         
    nl_rel_tol = 1.0e-8     
    nl_abs_tol = 1e-8       
    end_time = 0.01          
    [./TimeStepper]
      type = IterationAdaptiveDT  
      dt = 0.00005                 
      cutback_factor = 0.7        
      growth_factor = 1.2         
    [../]
[]  

[Adaptivity]
   initial_steps = 3                   
   max_h_level = 3                     
   initial_marker = err_phi            
  [./Markers]                          
     [./err_phi]
       type = ErrorFractionMarker      
       coarsen = 0.3
       refine = 0.95
       indicator = ind_phi
     [../]
  [../]
[] 

[Outputs]
    exodus = true
    interval = 10     
[]

[Debug]
  show_material_props = true
[]`

Is there any wrong with my input? Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

This ParsedMaterial is using A
```
[./mobility]
type = ParsedMaterial
coupled_variables = 'w'
f_name = tau
material_property_names = 'tau0 Le MCi k A'
expression = 'tau0A^2*(1/Le+MCi*(1+(1-k)w))'
output=exodus
[../]
``` 
but A is not defined. A below is the name of the Material not the material property. You need to find the actual name of the material property, either looking at the exodus output of the `show_material_props`
```
[Materials]
[./A]
type = InterfaceOrientationMaterial
op = phi
...
```

**Reply by Johnney12138** [reply]
Hi @GiudGiud 
Thank you for your reply.
In InterfaceOrientationMaterial, I didn't find an example of naming this block, such as material_property_names, etc.
The exodus output of the show_material_props is
`The following total 8 aux variables:
  ddepsdgrad_op_x
  ddepsdgrad_op_y
  ddepsdgrad_op_z
  deps
  depsdgrad_op_x
  depsdgrad_op_y
  depsdgrad_op_z
  eps
are added for automatic output by MaterialOutputAction.

Active Materials:
    Subdomain:  (0)
      Material Name:       A
      Property Names:      "ddepsdgrad_op" "deps" "depsdgrad_op" "eps" 
      Material Name:       speexpress
      Property Names:      "Fr" 
      Material Name:       mobility
      Property Names:      "tau" 
      Material Name:       chemp
      Property Names:      "chi" 
      Material Name:       matdiff
      Property Names:      "Dchi" 
      Material Name:       antitrap
      Property Names:      "rhodiff" 
      Material Name:       int
      Property Names:      "ft" 
      Material Name:       const
      Property Names:      "D" "L" "Le" "MCi" "erp" "k" "lambda" "tau0" "wid" 

    Active Face Materials:
    Subdomain:  (0)
      Material Name:       A_face
      Property Names:      "ddepsdgrad_op" "deps" "depsdgrad_op" "eps" 
      Material Name:       speexpress_face
      Property Names:      "Fr" 
      Material Name:       mobility_face
      Property Names:      "tau" 
      Material Name:       chemp_face
      Property Names:      "chi" 
      Material Name:       matdiff_face
      Property Names:      "Dchi" 
      Material Name:       antitrap_face
      Property Names:      "rhodiff" 
      Material Name:       int_face
      Property Names:      "ft" 
      Material Name:       const_face
      Property Names:      "D" "L" "Le" "MCi" "erp" "k" "lambda" "tau0" "wid" 

Active Neighboring Materials:
    Subdomain:  (0)
      Material Name:       A_neighbor
      Property Names:      "ddepsdgrad_op" "deps" "depsdgrad_op" "eps" 
      Material Name:       speexpress_neighbor
      Property Names:      "Fr" 
      Material Name:       mobility_neighbor
      Property Names:      "tau" 
      Material Name:       chemp_neighbor
      Property Names:      "chi" 
      Material Name:       matdiff_neighbor
      Property Names:      "Dchi" 
      Material Name:       antitrap_neighbor
      Property Names:      "rhodiff" 
      Material Name:       int_neighbor
      Property Names:      "ft" 
      Material Name:       const_neighbor
      Property Names:      "D" "L" "Le" "MCi" "erp" "k" "lambda" "tau0" "wid" 

Active Boundary Materials: `

**Reply by GiudGiud** [reply]
so the A you are using defines these
Property Names: "ddepsdgrad_op" "deps" "depsdgrad_op" "eps"

any of these is correct in the parsed expression you have?

**Comment by Johnney12138** [comment]
Hi @GiudGiud 
Thank you very much for your guidance and help. That is a stupid question. I looked at the InterfaceOrientationMaterial.C file and solved this problem.
Thanks again for your help.
Best,
Nan Liu
